bpf/process/types/basic.h

```diff
@@ -2378,6 +2383,12 @@ read_call_arg(void *ctx, struct msg_generic_kprobe *e, int index, int type,
 	case iov_iter_type:
 		size = copy_iov_iter(ctx, orig_off, arg, argm, e, data_heap);
 		break;
+	case linux_binprm_type: {
+		struct linux_binprm *bprm = (struct linux_binprm *)arg;
+		arg = (unsigned long)(&bprm->file);
+
+		// fallthrough to file_ty
```

Comment:
I don't believe that this is correct here. You expect to go through `linux_binprm_type`, `file_ty`, and `path_ty` but as it is now, my understanding is that it goes through `linux_binprm_type`, `kiocb_type`, `file_ty`, and `path_ty`. `kiocb_type` affects the value of arg which can cause issues.

Reply:
Thanks! I've switched this around a bit to be (hopefully) correct now!